engine/src/exec-interface-graphic.cpp

```diff
@@ -713,18 +733,47 @@ void MCGraphic::GetMarkerPoints(MCExecContext& ctxt, uindex_t& r_count, MCPoint*
 
 void MCGraphic::SetMarkerPoints(MCExecContext& ctxt, uindex_t p_count, MCPoint* p_points)
 {
```

Comment:
I suggest this for this method:
```

if (p_count > 65535)
{
	ctxt.LegacyThrow(EE_GRAPHIC_TOOMANY_POINTS)
	return
}

if (p_count == 0)
{
	flags &= ~F_MARKER_DRAWN;
	delete[] markerpoints;
	nmarkerpoints = 0;
}
else
{	
	MCPoint *t_closed_points;
	uindex_t t_r_count;
	uint2 t_newcount;
	DoCopyPoints(txt, p_count, p_counts, t_r_count, t_closed_points)
	t_newcount = uint2(t_r_count);
	if (flags & F_MARKER_OPAQUE)
	{
		if (!closepolygon(t_closed_points, t_newcount))
		{
			delete[] t_closed_points;
			ctxt.LegacyThrow(EE_GRAPHIC_TOOMANY_POINTS)
			return
		}
	}
	delete[] marker points
	markerpoints = t_closed_points;
	nmarkerpoints = t_newcount;
	flags |= F_MARKER_DRAWN
}
```
I think it is equivalent - it makes things a bit clearer as it illustrates better what is happening. i.e. If too many points to start with, throw. If new point count is zero, then turn off marker point drawing; otherwise, if markerOpaque then try and close the marker pointers, throwing if failure; then set the new markerpoints and set F_MARKER_DRAWN.

Note: There is a bug here due to closepolygon - it seems that closepolygon only works if F_OPAQUE is set (i.e. the graphic is opaque, so if F_OPAQUE is unset but F_MARKER_OPAQUE is set, the markerpoints *will not* get closed)